File: Android/Libraries/fireflydevice/src/test/java/com/fireflydesign/fireflydevice/FDCryptoTest.java
package com.fireflydesign.fireflydevice;

import junit.framework.Assert;
import junit.framework.TestCase;

import java.util.Arrays;

public class FDCryptoTest extends TestCase {

    public void testSha1() throws Exception {
        byte[] actual = FDCrypto.sha1(new byte[] {(byte)0x00, (byte)0x01, (byte)0x02, (byte)0x03, (byte)0x04, (byte)0x05, (byte)0x06, (byte)0x07, (byte)0x08, (byte)0x09, (byte)0x0a, (byte)0x0b, (byte)0x0c, (byte)0x0d, (byte)0x0f});
        byte[] expected = new byte[] {(byte)0xd4, (byte)0x8a, (byte)0xa2, (byte)0x4e, (byte)0x9f, (byte)0xee, (byte)0x0d, (byte)0xe3, (byte)0x40, (byte)0xea, (byte)0x7c, (byte)0xd5, (byte)0x13, (byte)0x88, (byte)0x6e, (byte)0xf6, (byte)0xe3, (byte)0x20, (byte)0xe9, (byte)0x09};
        Assert.assertTrue(Arrays.equals(actual, expected));
    }

}

File: FireflyDeviceTests/FDCryptoTest.m
//
//  FDCryptoTest.m
//  FireflyDevice
//
//  Created by Denis Bohm on 4/27/15.
//  Copyright (c) 2015 Firefly Design. All rights reserved.
//

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>

#import "FDCrypto.h"

@interface FDCryptoTest : XCTestCase

@end

@implementation FDCryptoTest

- (void)setUp {
    [super setUp];
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testSha1 {
    uint8_t inputBytes[] = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0f};
    NSData *input = [NSData dataWithBytes:inputBytes length:sizeof(inputBytes)];
    NSData *actual = [FDCrypto sha1:input];
    uint8_t expectedBytes[] = {0xd4, 0x8a, 0xa2, 0x4e, 0x9f, 0xee, 0x0d, 0xe3, 0x40, 0xea, 0x7c, 0xd5, 0x13, 0x88, 0x6e, 0xf6, 0xe3, 0x20, 0xe9, 0x09};
    NSData *expected = [NSData dataWithBytes:expectedBytes length:sizeof(expectedBytes)];
    XCTAssert([actual isEqualToData:expected]);
}

- (void)testPerformanceExample {
    // This is an example of a performance test case.
    [self measureBlock:^{
        // Put the code you want to measure the time of here.
    }];
}

@end
